### ISSUE TITLE
Exception while AmmySidekick.Ammy.Register, AddressAlreadyInUse

### ISSUE THREAD

**Issue by isukces** [AmmyUI/AmmyUI#72]
AmmySidekick.Ammy.Register causes Exception (SocketErrorCode, AddressAlreadyInUse, System.Net.Sockets.SocketError) under VisualStudio when another application using Ammy is running outside VS.
Another sympthoms (probably the same problems) are:
1. Unable to run compiled program (exe file) based on Ammy when another project using Ammy is running under VS.
2. Unable to run two instances of compiled program (exe file) based on Ammy

**Comment by ionoy** [comment]
Yep, this is a problem with a current implementation of Runtime Update. IDE uses a predefined port to communicate with a program. If two applications run concurrently, then one of them will fail to bind to a socket.

Ammy could fail gracefully and show a MessageBox at application start saying that Runtime Update will be disabled. 

You can also use "NO_AMMY_UPDATE" compilation symbol to disable runtime update (no exception will be thrown in this case.

Also, VS2017 natively supports runtime update, so Ammy doesn't always need it.

**Comment by isukces** [comment]
NO_AMMY_UPDATE is really helpful. I also read [issue #41].
